src/keyword_vpd_parser.cpp

```diff
@@ -0,0 +1,143 @@
+#include "keyword_vpd_parser.hpp"
+
+#include "constants.hpp"
+#include "exceptions.hpp"
+
+#include <iostream>
+#include <numeric>
+#include <string>
+
+namespace vpd
+{
+
+/* @brief Keyword VPD parser function*/
+std::variant<types::ParsedVPD, types::KeywordVpdMap> KeywordVpdParser::parse()
+{
+    try
+    {
+    m_vpdIterator = m_keywordVpdVector.begin();
+
+    if(*m_vpdIterator != constants::KW_VPD_START_TAG)
+    {
+        throw(DataException("Invalid Large resource type Identifier String"));
+    }
+
+    checkNextBytesValidity(sizeof(constants::KW_VPD_START_TAG));
+    std::advance(m_vpdIterator, sizeof(constants::KW_VPD_START_TAG));
+
+    uint16_t dataSize = getKwDataSize();
+
+    checkNextBytesValidity(constants::TWO_BYTES + dataSize);
+    std::advance(m_vpdIterator, constants::TWO_BYTES + dataSize);
+
+    // Check for invalid vendor defined large resource type
+    if (*m_vpdIterator != constants::KW_VPD_START_TAG)
+    {
+        if (*m_vpdIterator != constants::ALT_KW_VPD_START_TAG)
+        {
+            throw(DataException("Invalid Keyword Vpd Start Tag"));
```

Comment:
What needs to be done in case if these exceptions? I don't see them being handled anywhere.

Reply:
will look into it